lib/libmbedtls/core/rsa.c

```diff
@@ -22,6 +22,8 @@ static TEE_Result get_tee_result(int lmd_res)
 	switch (lmd_res) {
 	case 0:
 		return TEE_SUCCESS;
+	case MBEDTLS_ERR_RSA_PRIVATE_FAILED +
+		MBEDTLS_ERR_MPI_BAD_INPUT_DATA:
```

Comment:
How does this work? Do you mean that one of the MbedTLS functions return the actual sum of `MBEDTLS_ERR_RSA_PRIVATE_FAILED` and `MBEDTLS_ERR_MPI_BAD_INPUT_DATA`?
If so can point to an example in the MbedTLS code?

Reply:
I was surprised but It does, yes.
https://github.com/OP-TEE/optee_os/blob/3.13.0/lib/libmbedtls/mbedtls/library/rsa.c#L751-L770

Reply:
Wow! Here's a bit more information on the subject https://tls.mbed.org/api/error_8h.html
This is a bit fragile. What we really care about here is the `MBEDTLS_ERR_RSA_PRIVATE_FAILED` part of the error code.

Reply:
not that straight forward: the code is (-0x4300) + (-0x0004).

  (-0x4300) + (-0x0004)
= 0xffffffffffffbd00 + 0xfffffffffffffffc
= 0xffffffffffffbcfc

Reply:
The is the `mbedtls_strerror()` way in `lib/libmbedtls/mbedtls/library/error.c`

Reply:
I'm sorry, I didn't quite catch your meaning, mbedtls_strerror seems to be converting the errorno to a string. I need converting errorno (-0x4300) + (-0x0004) to TEE_ERROR_BAD_PARAMETERS. Is the way I'm doing it OK,Or is there any other way to modify it?

Reply:
Let's keep it as it is for now.

Reply:
OK，and you acked. @jenswi-linaro